mlir/lib/Target/LLVMIR/Dialect/GENX/GENXToLLVMIRTranslation.cpp

```diff
@@ -72,62 +72,46 @@ class GENXDialectLLVMIRTranslationInterface
 
     llvm::LLVMContext &llvmContext = moduleTranslation.getLLVMContext();
     llvm::Function *llvmFunc = moduleTranslation.lookupFunction(func.getName());
+    StringAttr attrName = attribute.getName();
+    Attribute attrVal = attribute.getValue();
 
     // Set calling convention for kernel
-    if (attribute.getName() ==
-        GENX::GENXDialect::getKernelFuncAttrName()) {
+    if (attrName == GENX::GENXDialect::getKernelFuncAttrName())
       llvmFunc->setCallingConv(llvm::CallingConv::SPIR_KERNEL);
-    }
-
-    // Set max_work_group_size metadata.
-    if (attribute.getName() ==
-        GENX::GENXDialect::getMaxWorkGroupSizeAttrName()) {
-      auto value = attribute.getValue().dyn_cast<ArrayAttr>();
-      if (!value)
-        return failure();
 
+    auto attachMetadata = [&](StringRef name) {
```

Comment:
Added a lambda to common up code used to set the attributes.

Reply:
Nice!